lib/sla.js

```diff
@@ -1,6 +1,77 @@
-'use strict';
+/*jslint node: true */
+'use strict'
 
-module.exports = function(callback){
-    console.log("\"snarflegluff\" should be printed below:");
-    var str = "snarflegluff";
-    return str;}
+/**
+ * Looks for an existing mixing bridge and creates one if none exist.
+ */
+function designateMixingBridge(client, callback) {
+  client.bridges.list(function(err, bridges) {
+    var bridge = bridges.filter(function(candidate) {
+      return candidate.bridge_type === 'mixing';
+    })[0];
+    if (bridge) {
+      console.log('Using existing mixing bridge %s', bridge.id);
+      createChannel(client, bridge, callback);
+    } else {
+      client.bridges.create({type: 'mixing'}, function(err, bridge) {
+        if (err) {
+          callback(err);
```

Comment:
Just as an FYI, calling a callback does not exit the function at this point. Since you are not returning from the function, the call to createChannel will occur even though an error. This means that createChannel will eventually error out and call the callback yet again, which is very bad. This is another thing that is bad about callbacks; even though they should only be called once, it's very easy to accidentally call them multiple times.